starport/services/scaffolder/type.go

```diff
@@ -13,45 +12,97 @@ import (
 	"github.com/tendermint/starport/starport/pkg/xgenny"
 	modulecreate "github.com/tendermint/starport/starport/templates/module/create"
 	"github.com/tendermint/starport/starport/templates/typed"
+	"github.com/tendermint/starport/starport/templates/typed/basic"
 	"github.com/tendermint/starport/starport/templates/typed/indexed"
 	"github.com/tendermint/starport/starport/templates/typed/singleton"
 )
 
-type TypeModel string
+// AddTypeOption configures options for AddType.
+type AddTypeOption func(*addTypeOptions)
 
-const (
-	List      TypeModel = "list"
-	Map       TypeModel = "map"
-	Singleton TypeModel = "singleton"
-)
+type addTypeOptions struct {
+	moduleName string
+	fields     []string
+
+	isList      bool
+	isMap       bool
+	isSingleton bool
+
+	index string
+
+	withoutMessage bool
+}
+
+// ListType enables adding a list storage.
+func ListType() AddTypeOption {
+	return func(o *addTypeOptions) {
+		o.isList = true
+	}
+}
+
+// MapType enables adding a map storage.
+func MapType(index string) AddTypeOption {
+	return func(o *addTypeOptions) {
+		o.isMap = true
+		o.index = index
+	}
+}
+
+// SingletonType enables adding a singleton storage.
+func SingletonType() AddTypeOption {
+	return func(o *addTypeOptions) {
+		o.isSingleton = true
+	}
+}
 
-type AddTypeOption struct {
-	Model     TypeModel
-	NoMessage bool
+// TypeWithModule specifies module to scaffold type inside.
+func TypeWithModule(name string) AddTypeOption {
+	return func(o *addTypeOptions) {
+		o.moduleName = name
+	}
 }
 
-// AddType adds a new type stype to scaffolded app by using optional type fields.
+// TypeWithFields adds fields to the type to be scaffold.
+func TypeWithFields(fields ...string) AddTypeOption {
+	return func(o *addTypeOptions) {
+		o.fields = fields
+	}
+}
+
+// TypeWithoutMessage disables CRUD for type.
```

Comment:
"prevent generating the messages for CRUD capability"

We still have the methods in the keeper that allows CRUD capability

Reply:
solved by 9204e56.